run/helm/snap/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: snap
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: snap
+      labels:
+        daemon: snapteld
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        securityContext:
+          privileged: {{ .Values.privileged }}
+        ports:
+        - containerPort: 8181
+        volumeMounts:
+        - name: autoload
+          mountPath: {{ .Values.config.autodiscover_path }}
+        - name: config
+          mountPath: /etc/snap
+        {{- range $key, $val := .Values.mounts }}
+        - name: {{ $key | replace "/" "" | replace "." "" }}
+          mountPath: {{ $val | quote }}
+        {{- end }}
+      volumes:
+      - name: autoload
+        hostPath:
```

Comment:
I know the problem that @jose5918 found was that the plugin autoload directory is on a read only host path when we try to start it up. I think we both want the same goal here of avoiding having to download plugins each time. The problem with using `hostPath` is that we are just faking persistent storage anyway and it is prone to just as many issues. In our case, the path for the autoload directory was read-only. There are also [other problems/considerations](https://kubernetes.io/docs/user-guide/volumes/#hostpath) for using a host path. But in all honesty, there isn't really an elegant solution we've thought of yet. Using host paths just feels hacky though.